compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2206,6 +2221,8 @@ private ParserRuleContext getNextRuleForTypeDescriptor() {
                 return ParserRuleContext.VARIABLE_NAME;
             case ANNOTATION_DECL:
                 return ParserRuleContext.IDENTIFIER;
+            case KEY_TYPE_CONSTRAINT_RHS:
+                return ParserRuleContext.GT;
```

Comment:
Similar to LT, if we can give an alias to `>`, we can avoid the complication inside `getNextRuleForGt`